pkg/datapath/link/link.go

```diff
@@ -61,3 +80,40 @@ func GetIfIndex(ifName string) (uint32, error) {
 	}
 	return uint32(iface.Attrs().Index), nil
 }
+
+type ifNameCache struct {
+	mu          lock.RWMutex
+	indexToName map[int]string
+}
+
+func (c *ifNameCache) syncCache() error {
+	indexToName := map[int]string{}
```

Comment:
Minor nit: this could be preallocated with `len(links)` before the `for _, link := range links` loop below:

```go
indexToName := make(map[int]string, len(links))
```